pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -240,8 +213,17 @@ func (c *Controller) syncOwnedProvisionRequest(ctx context.Context, wl *kueue.Wo
 		attempt := int32(1)
 		shouldCreatePr := false
 		if exists {
-			attempt = getAttempt(ctx, oldPr, wl.Name, checkName)
-			if apimeta.IsStatusConditionTrue(oldPr.Status.Conditions, autoscaling.Failed) {
+			switch {
+			case isCapacityRevoked(oldPr):
+				if workload.IsActive(wl) && !workload.IsFinished(wl) {
+					workload.SetEvictedCondition(wl, kueue.WorkloadEvictedByDeactivation, fmt.Sprintf("Deactivating workload because capacity for %v has been revoked", oldPr.Name))
+					if err := c.deactivateWorkload(ctx, wl); err != nil {
+						return nil, err
+					}
+					c.record.Eventf(wl, corev1.EventTypeNormal, "CapacityRevoked", "Deactivating workload because capacity for %v has been revoked", oldPr.Name)
```

Comment:
This seems like a warning-level event. It suggests something is misconfigured.

Reply:
Could you elaborate please? I thought it won't be treated as a warning-level event since its type is `Normal` instead of `Warning`

Reply:
```suggestion
					c.record.Eventf(wl, corev1.EventTypeWarning, "CapacityRevoked", "Deactivating workload because capacity for %v has been revoked", oldPr.Name)
```

Reply:
Yeah, changing to `Warning` is what I wanted to suggest.